solidity/contracts/bridge/Bridge.sol

```diff
@@ -306,34 +367,398 @@ contract Bridge is Ownable {
+        // be added because of how `parseVarInt` returns the length of the
+        // compactSize uint. Refer `BTCUtils` library for more details.
+        uint256 inputStartingIndex = 1 + inputsCompactSizeUintLength;
+
+        // Determine the swept deposits count. If the `previousSweepFlag` is
+        // already `true` that means the current sweep is the first one
+        // for given wallet and we shouldn't expect an additional input. If
+        // so, all inputs should be deposits. Otherwise, we need to subtract
+        // one input since it represents a previous sweep output.
+        depositors = new address[](
+            previousSweepFlag ? inputsCount : inputsCount - 1
+        );
+        depositedAmounts = new uint256[](depositors.length);
+
+        // Initialize helper variables.
+        uint256 processedDepositsCount = 0;
+
+        // Inputs processing loop.
+        for (uint256 i = 0; i < inputsCount; i++) {
+            // Check if we are at the end of the input vector.
+            if (inputStartingIndex >= sweepTxInputVector.length) {
+                break;
+            }
+
+            (bytes32 inputTxHash, uint32 inputTxIndex, uint256 inputLength) =
+                parseTxInputAt(sweepTxInputVector, inputStartingIndex);
+
+            DepositInfo storage deposit =
+                deposits[
+                    uint256(
+                        keccak256(abi.encodePacked(inputTxHash, inputTxIndex))
+                    )
+                ];
+
+            if (deposit.revealedAt != 0) {
+                require(deposit.sweptAt == 0, "Deposit already swept");
+
+                if (processedDepositsCount == depositors.length) {
+                    // If this condition is true, that means a deposit input
+                    // took place of an expected previous sweep input.
+                    // In that case, we should ignore that input and let the
+                    // transaction fail at `require(previousSweepFlag)`.
+                    continue;
+                }
+
+                /* solhint-disable-next-line not-rely-on-time */
+                deposit.sweptAt = uint32(block.timestamp);
+
+                depositors[processedDepositsCount] = deposit.depositor;
+                depositedAmounts[processedDepositsCount] = deposit.amount;
+                inputsValue += depositedAmounts[processedDepositsCount];
+
+                processedDepositsCount++;
+            } else if (
+                !previousSweepFlag && previousSweep.txHash == inputTxHash
+            ) {
+                inputsValue += previousSweep.txOutputValue;
+                previousSweepFlag = true;
+            } else {
+                revert("Unknown input type");
+            }
+
+            // Make the `inputStartingIndex` pointing to the next input by
+            // increasing it by current input's length.
+            inputStartingIndex += inputLength;
+        }
+
```

Comment:
Should we allow the wallet to make a sweep transaction that doesn't have any deposit inputs?
Such transaction would only burn some amount of BTC form the main UTXO (needed to pay for tx fee).
Perhaps we could add a check that `processedDepositsCount` is greater than `0`?

Reply:
Very good catch! That behavior is actually a fraud. Added a check: https://github.com/keep-network/tbtc-v2/pull/106/commits/b1c06a9499177b81c124704d51fd9ebcc5f39f73